build/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM golang:1.19-alpine as builder
+
+ENV GO111MODULE=on
+ENV CGO_ENABLED=0
+WORKDIR /work
+
+ADD go.mod go.sum /work/
+RUN go mod download
```

Comment:
question (non-blocking): am I correct in understanding that you download modules separately to better leverage build caching?

So that the layers stay cached as long as dependencies stay the same, and cache is still valid. If so, then I’m stealing this trick :grin:

Reply:
yes, but build still takes ages given all deps we need to recompile each time
there's even better solutions: https://github.com/edwarnicke/imports-gen

Reply:
Thanks!